### PR TITLE
Added GPU failure regex

### PR DESCRIPTION
**Pull request by Navidem**
Added the testcase based on this [report](https://clusterfuzz.com/testcase-detail/5766916010213376).
I am not sure if the `new_state` and `new_type` values are suitable.

### CODE REVIEW

File: src/clusterfuzz/stacktraces/__init__.py [hunk 1]
```diff
@@ -814,6 +814,15 @@ def parse(self, stacktrace: str) -> CrashInfo:
           state,
           new_type='Kernel failure\nGeneral-protection-fault')
 
+      # GPU Failure.
+      self.update_state_on_match(
+        GPU_PROCESS_FAILURE,
```
Comment (inferno-chromium) [inline src/clusterfuzz/stacktraces/__init__.py:819]: i think you need to run python butler.py format to fix indentation (see ci/cd failure).

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py [hunk 2]
```diff
@@ -193,6 +193,18 @@ def test_android_asan_unknown_write(self):
...
                                   expected_security_flag)
 
+  def test_gpu_failure(self):
+    """Basic test for GPU failure."""
+    data = self._read_test_data('gpu_failure.txt')
+    expected_type = 'GPU failure'
+    expected_address = ''
+    expected_stacktrace = data
+    expected_state = 'NULL'
+    expected_security_flag = True
```
Comment (inferno-chromium) [inline src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:203]: Are the tests passing, i think you need to make a crash_analyzer change to make this crash type "GPU failure" security one. see https://github.com/google/clusterfuzz/blob/4f8020c4c7ce73c1da0e68f04943af30bb5f0b32/src/clusterfuzz/_internal/crash_analysis/crash_analyzer.py#L348

**Review by inferno-chromium — CHANGES_REQUESTED** [review]

File: src/clusterfuzz/stacktraces/__init__.py (hunk 1 above)
Comment (Navidem) [reply src/clusterfuzz/stacktraces/__init__.py]: Done.

File: src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py (hunk 2 above)
Comment (Navidem) [reply src/clusterfuzz/_internal/tests/core/crash_analysis/stack_parsing/stack_analyzer_test.py:203]: Tests were passing on my end. I will double check.

**Comment by inferno-chromium** [comment]
Ah right, new crash type automatically becomes security. Can ignore then.

**Review by inferno-chromium — APPROVED** [review]

**Comment by inferno-chromium** [comment]
@oliverchang - can you please help with deployment.

**Comment by oliverchang** [comment]
> @oliverchang - can you please help with deployment.

Deploying!

**Pull request merged**